Lib/fontTools/varLib/merger.py

```diff
@@ -766,8 +778,10 @@ def merge(merger, self, lst):
 		if not sts:
 			continue
 		if sts[0].__class__.__name__.startswith('Extension'):
-			assert allEqual([st.__class__ for st in sts])
-			assert allEqual([st.ExtensionLookupType for st in sts])
+			if not allEqual([st.__class__ for st in sts]):
+				raise VarLibMergeError()
```

Comment:
the problem here is when, while merging a lookup that contains Extension subtables, not all the masters use extensions or have the same extension lookup type (because later it is assuming the first subtable lookup type is the one to be used for all). I know it's hard to come up with a meaningful error message here. But raising a muted exception with no error is no better than before.